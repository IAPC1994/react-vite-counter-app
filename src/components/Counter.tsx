import { useState } from 'react'

interface Props {
  initialValue: number;
}

const Counter = ({ initialValue }:Props) => {
  const [count, setCount] = useState(initialValue);

  const increment = () => {
    setCount( count + 1 );
  }
  const decrement = () => {
    setCount( count - 1 );
  }
  const reset = () => {
    setCount( initialValue ); 
  }
  const switchSigns = () => {
    setCount( count * -1 );
  }

  return (
    <div className='container'>
      <h1>
        Count: { count }
      </h1>
      <div>
        <button onClick={ increment }> Increment </button>
        <button onClick={ decrement }> Decrement </button>
        <button onClick={ reset }> Restart </button>
        <button onClick={ switchSigns }> Switch Signs </button>
      </div>
    </div>
  )
}

export default Counter
