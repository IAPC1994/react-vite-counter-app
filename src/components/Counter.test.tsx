import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';


describe('Counter App', () => { 
    
    // beforeEach(() => {
    //     render(<Counter initialValue={ 0 } />)
    // });

    const initialValue = 100;

    test('Should have match with the snapshot', () => {
        const { container } = render( <Counter initialValue={ 0 } />);
        expect( container ).toMatchSnapshot();
    })

    test('The initial value of the count should be 100', () => {
        render( <Counter initialValue={ initialValue } />);
        //screen.debug();
        expect( screen.getByText( 'Count: 100' )).toBeTruthy();
    })
    
    test('Increment the counter to 1', () => { 
        render( <Counter initialValue={ 0 } />);
        const increment = screen.getByText('Increment');
        fireEvent.click(increment);
        fireEvent.click(increment);

        expect(screen.getByText('Count: 2')).toBeTruthy();
    });

    test('Decrement the counter to -2', () => { 
        render( <Counter initialValue={ 0 } />);
        const decrement = screen.getByText('Decrement');
        fireEvent.click(decrement);
        fireEvent.click(decrement);

        expect(screen.getByText('Count: -2')).toBeTruthy();
    });

    test('Reset the counter should to be 100', () => { 
        render( <Counter initialValue={ initialValue } />);
        const reset = screen.getByText('Restart');
        const increment = screen.getByText('Increment');
        fireEvent.click(increment);
        fireEvent.click(increment);
        fireEvent.click(increment);

        fireEvent.click(reset);

        expect(screen.getByText('Count: 100')).toBeTruthy();
    });

    test('Switch signs should be negative', () => { 
        render( <Counter initialValue={ initialValue } />);
        const switchSigns = screen.getByText('Switch Signs');
        const increment = screen.getByText('Increment');
        fireEvent.click(increment);
        fireEvent.click(increment);

        fireEvent.click(switchSigns);

        expect(screen.getByText('Count: -102')).toBeTruthy();
    });
});