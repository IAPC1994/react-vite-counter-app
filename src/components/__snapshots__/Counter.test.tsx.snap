// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter App > Should have match with the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Count: 
      0
    </h1>
    <div>
      <button>
         Increment 
      </button>
      <button>
         Decrement 
      </button>
      <button>
         Restart 
      </button>
      <button>
         Switch Signs 
      </button>
    </div>
  </div>
</div>
`;
